### ISSUE TITLE
Autocomplete: Wrong name when using custom subcommand names

### ISSUE THREAD

**Issue by Parakoopa** [pallets/click#1213]
Hi! When using custom names for commands the autocomplete does not work correctly. It outputs the name of the function (`cmd`), not the name assigned in add_command (`my_custom_name`):
```python

@click.group()
def group():
    pass

@click.command()
def cmd():
    pass

group.add_command(cmd,    'my_custom_name')
```
I was able to fix the issue by overriding add_command in the following way:
```python
    def add_command(self, cmd, name=None):
        cmd.name = name
        return super().add_command(cmd, name)
```
However I don't know if this raises other issues or if this works in all cases.

**Comment by rossmacarthur** [comment]
Forgive me if I have misunderstood what you want, but the following is how you generally use subcommands in groups:

```python
@click.group()
def group():
    pass

@group.command('my_custom_name')  # NB: this is group.command() not click.command()
def cmd():
    pass
```

**Comment by kx-chen** [comment]
Still an issue in the new autocomplete system (https://github.com/pallets/click/pull/1622). 
![Screen Shot 2020-08-10 at 11 10 11 AM](https://user-images.githubusercontent.com/22359229/89815837-4acf4c00-dafa-11ea-80c8-d82d3d1eaee5.png)

Passing the name in the decorator works.
![Screen Shot 2020-08-10 at 11 10 48 AM](https://user-images.githubusercontent.com/22359229/89815886-66d2ed80-dafa-11ea-8e4b-aef54e66cf87.png)

**Comment by davidism** [comment]
This was fixed as part of #1622, specifically 00d5dcc.
